### PR TITLE
KAA-1248: [C++ SDK] Endpoint doesn't reconnect to Bootstrap server on start.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by rasendubi** [comment]
License headers are missing.

File: client/client-multi/client-cpp/impl/channel/impl/DefaultOperationTcpChannel.cpp [hunk 1]
```diff
@@ -217,14 +217,18 @@ void DefaultOperationTcpChannel::openConnection()
...
-    boost::asio::ip::tcp::endpoint ep;
-    try {
-        ep = HttpUtils::getEndpoint(currentServer_->getHost(), currentServer_->getPort());
-    } catch (std::exception& e) {
-        KAA_LOG_ERROR(boost::format("Channel \"%1%\". Connection to endpoint failed: %2%") % getId() % e.what());
+
+    boost::system::error_code errorCode;
+
+    boost::asio::ip::tcp::endpoint ep = HttpUtils::resolveEndpoint(currentServer_->getHost(),
+                                                               currentServer_->getPort(),
```
Comment (ohamov) [inline client/client-multi/client-cpp/impl/channel/impl/DefaultOperationTcpChannel.cpp]: Code style

File: client/client-multi/client-cpp/test/impl/channel/IPConnectivityCheckerTest.cpp [hunk 2]
```diff
@@ -19,68 +19,23 @@
...
     PingServerStorage(const std::string& host, const std::uint16_t& port) {
-        server_ = createTransportConnectionInfo(ServerType::BOOTSTRAP
-                                          , 0x111
-                                          , TransportProtocolIdConstants::HTTP_TRANSPORT_ID
-                                          , serializeConnectionInfo("key"
-                                                                  , host
-                                                                  , port));
+        server_ = KaaTestUtils::createTransportConnectionInfo(ServerType::BOOTSTRAP
+                                                            , 0x111
+                                                            , TransportProtocolIdConstants::HTTP_TRANSPORT_ID
```
Comment (ohamov) [inline client/client-multi/client-cpp/test/impl/channel/IPConnectivityCheckerTest.cpp]: Code style (commas)

File: client/client-multi/client-cpp/test/headers/KaaTestUtils.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+namespace kaa {
+
+class KaaTestUtils {
+public:
+    static std::vector<uint8_t> serializeConnectionInfo(const std::string& host,
+                                                        std::int32_t port,
+                                                        const PublicKey& publicKey);
+
+    static ITransportConnectionInfoPtr createTransportConnectionInfo(ServerType type,
+                                                                    std::int32_t accessPointId,
```
Comment (ohamov) [inline client/client-multi/client-cpp/test/headers/KaaTestUtils.hpp]: Code style

File: client/client-multi/client-cpp/impl/channel/impl/DefaultOperationTcpChannel.cpp (hunk 1 above)
Comment (ghost) [inline client/client-multi/client-cpp/impl/channel/impl/DefaultOperationTcpChannel.cpp]: Done.

File: client/client-multi/client-cpp/test/impl/channel/IPConnectivityCheckerTest.cpp (hunk 2 above)
Comment (ghost) [inline client/client-multi/client-cpp/test/impl/channel/IPConnectivityCheckerTest.cpp]: Done.

File: client/client-multi/client-cpp/test/headers/KaaTestUtils.hpp (hunk 3 above)
Comment (ghost) [inline client/client-multi/client-cpp/test/headers/KaaTestUtils.hpp]: Done.

**Comment by ohamov** [comment]
LGTM

**Pull request merged**